packages/core/src/operator/listenForChallenges.ts

```diff
@@ -11,41 +9,76 @@ import { resilientEventListener } from "../utils/resilientEventListener.js";
-
-            if(error){
-                //TODO hanlde on error
+            if (error) {
+                // Call the onError function with the caught error
+                onError(error instanceof Error ? error : new Error(String(error)));
                 return;
             }
+            try {
+                // Uncomment the following line to test the onError function
+                // throw new Error("Test Error: Simulating a failure in getChallenge");
+                const challengeNumber = BigInt(log?.args[0]);
 
-            const challengeNumber = BigInt(log?.args[0]);
-
-            // if the challengeNumber has not been seen before, call the callback and add it to the map
-            if (!challengeNumberMap[challengeNumber.toString()]) {
-                challengeNumberMap[challengeNumber.toString()] = true;
+                // if the challengeNumber has not been seen before, call the callback and add it to the map
+                if (!challengeNumberMap[challengeNumber.toString()]) {
+                    challengeNumberMap[challengeNumber.toString()] = true;
 
-                // lookup the challenge
-                const challenge = await getChallenge(challengeNumber);
+                    // lookup the challenge
+                    const challenge = await getChallenge(challengeNumber);
 
-                void callback(challengeNumber, challenge, log);
+                    void callback(challengeNumber, challenge, log);
+                }
+            } catch (err) {
+                // Call the onError function with the caught error
+                onError(err instanceof Error ? err : new Error(String(err)));
             }
         }
     });
 
-    // return a function that can be used to stop listening for the event
     return () => {
         listener.stop();
     };
 }
+
+
+//// Below only used to mock callback for testing error handler
+//// Define the type for the parameters of resilientEventListener
+// interface ResilientEventListenerParams {
+//     rpcUrl: string;
+//     contractAddress: string;
+//     abi: any[]; // Adjust the type as needed for your ABI
+//     eventName: string;
+//     log: (message: string) => void;
+//     callback: (log: LogDescription | null, error?: Error) => void;
+//   }
+  
+//   // Mock implementation of resilientEventListener
+//   const mockResilientEventListener = ({
+//     rpcUrl,
+//     contractAddress,
+//     abi,
+//     eventName,
+//     log,
+//     callback
+//   }: ResilientEventListenerParams): { stop: () => void } => {
+//     // Mock the stop function
+//     const stop = () => log("Mock listener stopped");
+  
+//     // Simulate an event being triggered
+//     setTimeout(() => {
+//       const mockLog: LogDescription = { name: 'MockEvent', args: ['arg1', 'arg2'] } as LogDescription; // Example log, cast as LogDescription
+//       callback(mockLog, undefined); // No error
+//     }, 2500); // 1-second delay for simulation
+  
+//     return { stop };
```

Comment:
Not a huge fan of having this commented code in here, any reason we don't put it into a readme and the ticket ?

Reply:
For testing you could also update the websocket uri in the config.ts and test with a local websocket server that sends error data, thats usually a more reliable way to test that adding additional code for mocking functions. Let me know what you think.

Reply:
Removed, and understood on the local websocket server, I've always just mocked for testing, but will consider this moving forward.